ledger-api/rs-grpc-akka/src/main/scala/com/digitalasset/grpc/adapter/server/akka/StreamingServiceLifecycleManagement.scala

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.grpc.adapter.server.akka
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Keep, Source}
+import akka.stream.{KillSwitch, KillSwitches, Materializer}
+import com.daml.grpc.adapter.ExecutionSequencerFactory
+import io.grpc.stub.StreamObserver
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.ExecutionContext
+
+trait StreamingServiceLifecycleManagement extends AutoCloseable {
+  @volatile private var _closed = false
+  private val _killSwitches = TrieMap.empty[KillSwitch, Object]
```

Comment:
Are the values from this map used for anything? Could we do just as well with a thread safe list?

Reply:
Ideally the collection should have set-like properties for the killSwitches (e.g. O(1) removal) which for lists we don't have. On the other hand, the TrieMap is the goto concurrent collection for such uses (`SharedKillSwitch` also uses it like this)